### PR TITLE
#170766078 Done with configuring swagger documentation

### PR DESCRIPTION
**Pull request by key-joshua**
## WHAT DOES THIS PR DO?

- I have configured swagger documentation tools.
- I have created a swagger file.

## DESCRIPTION OF TASK TO BE COMPLETED?

- During the completion of this work I have installed and configured swagger documentation
  and those swagger file will help the users to get documented on how application work.

####  The way this will be tested:

- Users should pull from this branch or clone this repository into his/her machine.
- User after clone this repository should run ```npm install``` command to install all required tools.
- Users will have to run ```npm run server``` command to start the server.
- Users should have to open his/her favorite browser which will be using to testing this documentation.
- Users should use this link (http://localhost:3000/api/v-unknown/documentation/) or could use any invalid link to be able redirected to the documentation.

## ANY BACKGROUND CONTEXT YOU  WANT TO PROVIDE?

- The context was all about nodejs with libraries.
- [Swagger documentation](https://swagger.io/tools/open-source/getting-started/)

## WHAT ARE THE RELEVANT PIVOTAL TRACKER STORIES?
- [Chore: Setup API documentation using swagger](https://www.pivotaltracker.com/story/show/170766078)

#### Screenshots (if appropriate)
swagger screen shot

### CODE REVIEW

**Comment by Nigorjeanluc** [comment]
It looks good for me

**Review by jajajaden01 — COMMENTED** [review]
Good job.

**Review by karengir — DISMISSED** [review]
good work!

**Review by izabayogit — COMMENTED** [review]
image

here change the message to show the user that he is on wrong endpoint, because welcome message means that the user is on right endpoint while status code shows bad request

**Review by izabayogit — DISMISSED** [review]

**Comment by key-joshua** [comment]
Thank you, Mr. Jonas I have done with your feedback and I hope you can make another review if you want.

**Review by jajajaden01 — DISMISSED** [review]
those configurations look good

**Review by karengir — DISMISSED** [review]
Good work!

**Comment by key-joshua** [comment]
Thank you guyz and let stay on touch !!

**Review by luc-tuyishime — COMMENTED** [review]
Fix the hound CI issues please.

**Review by Nigorjeanluc — DISMISSED** [review]

**Review by jajajaden01 — DISMISSED** [review]

**Review by karengir — DISMISSED** [review]

File: server/index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+const app = express();
+app.use(express.urlencoded({ extended: true }));
+app.use(express.json());
+const basePath = '/api/v-unknown';
+
+app.listen(process.env.PORT || 3000, () => {
+  console.log('server is running on port 3000');
+});
+app.use(`${basePath}/documentation`, swaggerUi.serve, swaggerUi.setup(apiDocumentation));
+app.get('**', (req, res) => {
```
Comment (Nigorjeanluc) [inline server/index.js:14]: I think you need to test this endpoint

**Review by Nigorjeanluc — COMMENTED** [review]
Your `server/index.js` needs to be tested

File: server/index.js (hunk 1 above)
Comment (key-joshua) [reply server/index.js:14]: Yeah definitely, this was tested from the chore am working on called Setup unit testing and after pushing this chore I will be having testing environment. with one test of that endpoint.

**Pull request merged**